src/SFA.DAS.AssessorService.Web.UnitTests/CertificateTests/CertificateControllerTests.cs

```diff
@@ -83,16 +93,24 @@ public async Task WhenStartingANewCertificate_WithSingleVersionNoOptions_GoesToD
         [Test, MoqAutoData]
         public async Task WhenStartingANewCertificate_WithMultipleVersions_GoesToVersionPage(CertificateStartViewModel model, IEnumerable<StandardVersion> standards)
         {
+            CertificateSession setSession = new CertificateSession();
             _mockStandardVersionClient.Setup(s => s.GetStandardVersionsByLarsCode(model.StdCode)).ReturnsAsync(standards);
-
+            _mockSessionService.Setup(c => c.Set("CertificateSession", It.IsAny<object>()))
+                .Callback<string, object>((key, session) =>
+                 {
+                     if (key == "CertificateSession" && session is CertificateSession)
+                     {
+                         setSession = (CertificateSession)session;
+                     }
+                 });
             var result = await _certificateController.Start(model) as RedirectToActionResult;
 
             _mockCertificateApiClient.Verify(s => s.Start(It.Is<StartCertificateRequest>(
                 m => m.StandardCode == model.StdCode && m.Uln == model.Uln
                 && m.Username == Username && m.UkPrn == Ukprn)));
 
-            _mockSessionService.Verify(c => c.Set("CertificateSession", It.Is<CertificateSession>(
-                s => s.CertificateId == CertificateId && s.Uln == model.Uln && s.StandardCode == model.StdCode)));
+            _mockSessionService.Verify(c => c.Set("CertificateSession", It.Is<CertificateSession>(s => s.CertificateId == CertificateId && s.Uln == model.Uln && s.StandardCode == model.StdCode)));
```

Comment:
`setSession` which is set in the above call back can have its values asserted rather than this verify. Much cleaner, and more straightforward when something goes wrong

Reply:
Yep that's fair. I was fighting with the verify to try and get it to verify the versions and so introduced the callback to capture correctly to allow an assert. Just didn't take it all the way and just remove the verify. Simplifies the multiple checks on the set to just confirming the final session object which simplifies the test.